howto/addons/create-addon.md

```diff
@@ -17,4 +21,113 @@ cat "$ADDON_DIR"/public_key.pem >> ~/.ssh/authorized_keys
 To create an addon, you must provide the Anbox Management Client (AMC) with either of the following:
 * The addon directory
 * A tarball containing the required addon file structure
-* A zip archive containing the required addon file structure
\ No newline at end of file
+* A zip archive containing the required addon file structure
+
+Let's look at an example for of an addon for enabling SSH access on a container.
+
+## Write the addon metadata
+
+In a new `ssh-addon` directory, create a `manifest.yaml` file with the following content:
+```yaml
+name: ssh
+description: |
+  Enable SSH access when starting a container
+```
+
+## Add a hook
+
+In the `ssh-addon` directory, create a `hooks` directory. Hooks can be implemented in any language, this example uses a bash script.
+
+In the `hooks` directory, create a `pre-start` file with the following content:
+
+```bash
+#!/bin/bash
+
+if [ "$INSTANCE_TYPE" = "regular" ]; then
+  exit 0
+fi
+
+mkdir -p ~/.ssh
+cat "$ADDON_DIR"/ssh-addon-key.pub >> ~/.ssh/authorized_keys
+```
+
+To make the `pre-start` file executable, run the following command in the `ssh-addon` directory:
+
+        chmod +x hooks/pre-start
+
+```{tip}
+- Supported hooks are `pre-start`, `post-start` and `post-stop`.
+- Use the `INSTANCE_TYPE` variable to distinguish between regular and base instances.
+
+See {ref}`ref-hooks` for more information.
+```
+Create an SSH key in your addon directory and move the private key to a location outside of the addon directory:
+
+        ssh-keygen -f ssh-addon-key -t ecdsa -b 521
+        mv ssh-addon-key ~/
+
+Alternatively, you can use an existing key and move the public key into the addon directory.
```

Comment:
And public key must be renamed as ssh-addon-key.pub if people are using the existing key?